packages/charts/src/chart_types/partition_chart/layout/utils/legend_labels.ts

```diff
@@ -32,15 +48,19 @@ function flatSlicesNames(
     const formattedValue = formatter ? formatter(key) : `${key}`;
     // preventing errors from external formatters
     if (formattedValue && formattedValue !== HIERARCHY_ROOT_KEY) {
+      // Node key must be unique for each node in the tree
+      const nodeKey = getArrayNodeKey(arrayNode);
       // save only the max depth, so we can compute the the max extension of the legend
-      keys.set(formattedValue, Math.max(depth, keys.get(formattedValue) ?? 0));
+      depths.set(nodeKey, Math.max(depth, depths.get(nodeKey) ?? 0));
```

Comment:
do we need to compute the `max` depth now since we are creating unique nodeKeys for each node?

Reply:
@nickofthyme from the previous line comment the nodeKey is unique for each node in the tree, so you probably can rewrite the current line with:
```suggestion
      depths.set(nodeKey, depth);
```